core/arch/riscv64/riscv64.asm

```diff
@@ -59,7 +59,25 @@ GLOBAL_LABEL(cpuid_supported:)
  */
         DECLARE_FUNC(call_switch_stack)
 GLOBAL_LABEL(call_switch_stack:)
-/* FIXME i#3544: Not implemented */
+        addi    sp, sp, -24 /* Init the stack */
+        sd      ra, 16 (sp)
+        sd      s0, 8 (sp)
+        sd      s1, 0 (sp)
+        /* Check mutex_to_free. */
+        beqz     ARG4, call_dispatch_alt_stack_no_free
+        /* Release the mutex. */
+        addi     ARG4,x0,0
```

Comment:
Comparing with the AArch64 version, we're storing zero in [ARG4] (`str      wzr, [ARG4]`) whereas here we seem to be storing zero in `ARG4`.

Reply:
X0 register is a zero register in RISC-V. 
So add a imm zero and a zero register equals zero.

Reply:
```
addi      ARG4, x0 ,0
                0  =  0 + 0
```

Reply:
I mean: do we need to store a zero in the address pointed by the ARG4 reg or in the ARG4 reg itself?

Reply:
Ok. In the new version, I sd a zero in the address pointed by ARG4 (with no offset)